src/Fcf/Data/List.hs

```diff
@@ -50,10 +54,49 @@ type instance Eval (Foldr f y (x ': xs)) = Eval (f x (Eval (Foldr f y xs)))
 data UnList :: b -> (a -> b -> Exp b) -> [a] -> Exp b
 type instance Eval (UnList y f xs) = Eval (Foldr f y xs)
 
+-- Helper for the Unfoldr.
+data UnfoldrCase :: (b -> Exp (Maybe (a,b))) -> Maybe (a,b) -> Exp [a]
+type instance Eval (UnfoldrCase f ('Just ab)) =
+  Eval (Fst ab) ': Eval (Unfoldr f (Eval (Snd ab)))
+type instance Eval (UnfoldrCase _ 'Nothing) = '[]
+
+-- | Type-level Unfoldr.
+--
+-- Example:
+--
+-- @
+-- data ToThree :: Nat -> Exp (Maybe (Nat,Nat))
+-- type instance Eval (ToThree b) = 
+--   If ( Eval ( b Fcf.>= 4) )
+--     'Nothing
+--     ( 'Just '(b, b TL.+ 1 ))
+--
+-- :kind! Eval (Unfoldr ToThree 0)
+-- @
+--
+-- See also the definition of `Replicate`.
+data Unfoldr :: (b -> Exp (Maybe (a,b))) -> b -> Exp [a]
+type instance Eval (Unfoldr f c) = Eval (UnfoldrCase f (f @@ c) )
+
 data (++) :: [a] -> [a] -> Exp [a]
 type instance Eval ((++) '[] ys) = ys
 type instance Eval ((++) (x ': xs) ys) = x ': Eval ((++) xs ys)
 
+-- | Concat for lists.
+--
+-- Examples:
+--
+-- @
+-- :kind! Eval (Concat ( '[ '[1,2], '[3,4], '[5,6])
```

Comment:
There's a missing closing bracket here

Reply:
outs, added closing bracket and closing parenthesis to this and the next one